flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -73,7 +76,14 @@ public KinesisProxy(Properties configProps) {
 		this.configProps = checkNotNull(configProps);
 
 		this.regionId = configProps.getProperty(KinesisConfigConstants.CONFIG_AWS_REGION);
+		ClientConfigurationFactory configurationFactory = new ClientConfigurationFactory();
+		ClientConfiguration config = configurationFactory.getConfig();
+		if(config.getUserAgent().equals(ClientConfiguration.DEFAULT_USER_AGENT)) {
```

Comment:
Why do we have to check if it equals the default user agent before setting it to Flink?